File: Source/ECHashExtensions.m
//
//  Digest.m
//  EasyCrypto
//
//  Created by Fausto Ristagno on 16/09/15.
//  Copyright © 2015 Fausto Ristagno. All rights reserved.
//

#import "ECDigest.h"
#import "ECDigestObject.h"

@implementation ECDigest
@end

// MARK: Extensions
@implementation NSString (ECDigest)

- (nonnull ECMd2 *)md2
{
    return [[ECMd2 alloc] initWithString:self];
}

- (nonnull ECMd4 *)md4
{
    return [[ECMd4 alloc] initWithString:self];
}

- (nonnull ECMd5 *)md5
{
    return [[ECMd5 alloc] initWithString:self];
}

- (nonnull ECSha1 *)sha1
{
    return [[ECSha1 alloc] initWithString:self];
}

- (nonnull ECSha224 *)sha224
{
    return [[ECSha224 alloc] initWithString:self];
}

- (nonnull ECSha256 *)sha256
{
    return [[ECSha256 alloc] initWithString:self];
}

- (nonnull ECSha384 *)sha384
{
    return [[ECSha384 alloc] initWithString:self];
}

- (nonnull ECSha512 *)sha512
{
    return [[ECSha512 alloc] initWithString:self];
}


@end

@implementation NSData (ECDigest)

- (nonnull ECMd2 *)md2
{
    return [[ECMd2 alloc] initWithData:self];
}

- (nonnull ECMd4 *)md4
{
    return [[ECMd4 alloc] initWithData:self];
}

- (nonnull ECMd5 *)md5
{
    return [[ECMd5 alloc] initWithData:self];
}

- (nonnull ECSha1 *)sha1
{
    return [[ECSha1 alloc] initWithData:self];
}

- (nonnull ECSha224 *)sha224
{
    return [[ECSha224 alloc] initWithData:self];
}

- (nonnull ECSha256 *)sha256
{
    return [[ECSha256 alloc] initWithData:self];
}

- (nonnull ECSha384 *)sha384
{
    return [[ECSha384 alloc] initWithData:self];
}

- (nonnull ECSha512 *)sha512
{
    return [[ECSha512 alloc] initWithData:self];
}


@end

File: Source/ECHashExtensions.swift
//
//  ECDigestExtensions.swift
//  EasyCrypto
//
//  Created by Fausto Ristagno on 16/09/15.
//  Copyright © 2015 Fausto Ristagno. All rights reserved.
//

public extension String {
    func md2() -> ECHashMD2 {
        return (self as NSString).md2()
    }
    
    func md4() -> ECHashMD4 {
        return (self as NSString).md4()
    }
    
    func md5() -> ECHashMD5 {
        return (self as NSString).md5()
    }
    
    func sha1() -> ECHashSHA1 {
        return (self as NSString).sha1()
    }
    
    func sha224() -> ECHashSHA224 {
        return (self as NSString).sha224()
    }
    
    func sha256() -> ECHashSHA256 {
        return (self as NSString).sha256()
    }
    
    func sha384() -> ECHashSHA384 {
        return (self as NSString).sha384()
    }
    
    func sha512() -> ECHashSHA512 {
        return (self as NSString).sha512()
    }
}
